crates/holochain_conductor_config/src/generate.rs

```diff
@@ -0,0 +1,169 @@
+
+/// Generate a new directory structure for configurations
+pub fn generate_config_directory(
+    root: Option<PathBuf>,
+    directory: Option<PathBuf>,
+) -> anyhow::Result<ConfigRootPath> {
+    let mut dir = root.unwrap_or_else(std::env::temp_dir);
+    let directory = directory.unwrap_or_else(|| nanoid::nanoid!().into());
+    dir.push(directory);
+    std::fs::create_dir(&dir)?;
+
+    Ok(dir.into())
+}
+
+pub fn init_lair(dir: &KeystorePath, passphrase: sodoken::BufRead) -> anyhow::Result<url2::Url2> {
+    match init_lair_inner(dir, passphrase) {
+        Ok(url) => Ok(url),
+        Err(err) => Err(std::io::Error::new(
+            std::io::ErrorKind::Other,
+            format!("Failed to execute 'lair-keystore init': {:?}", err),
+        )
+        .into()),
+    }
+}
+
+pub(crate) fn init_lair_inner(
+    dir: &KeystorePath,
+    passphrase: sodoken::BufRead,
+) -> anyhow::Result<url2::Url2> {
+    let mut cmd = std::process::Command::new("lair-keystore");
+
+    cmd.args(["init", "--piped"])
+        .current_dir(dir.as_ref())
+        .stdin(std::process::Stdio::piped());
+
+    let mut proc = cmd.spawn()?;
+    let mut stdin = proc.stdin.take().unwrap();
+
+    use std::io::Write;
+    stdin.write_all(&passphrase.read_lock()[..])?;
+    stdin.flush()?;
+    drop(stdin);
+
+    if !proc.wait()?.success() {
+        return Err(std::io::Error::new(std::io::ErrorKind::Other, "LairInitFail").into());
+    }
+    let conf = dir.as_ref().join("lair-keystore-config.yaml");
+
+    let conf = std::fs::read(conf)?;
+
+    #[derive(serde::Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    struct Conf {
+        connection_url: url2::Url2,
+    }
+
+    let conf: Conf = serde_yaml::from_slice(&conf)?;
+
+    Ok(conf.connection_url)
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::config::read_config;
+
+    use anyhow::Context;
+    use holochain_conductor_api::conductor::KeystoreConfig;
+    use kitsune_p2p_types::config::{KitsuneP2pConfig, KitsuneP2pTuningParams, TransportConfig};
+    use tempfile::tempdir;
+
+    #[test]
+    fn test_generate_creates_default_config_file() -> anyhow::Result<()> {
+        let temp_dir = tempdir()?;
+        let root = Some(temp_dir.path().to_path_buf());
+        let directory = Some("test-config".into());
+
+        let config_root = generate(None, root, directory, true, 0)?;
+
+        assert!(config_root.as_path().exists());
+        assert!(config_root.as_path().is_dir());
+
+        let config_file = config_root.as_path().join("conductor-config.yaml");
+        assert!(config_file.exists());
+        assert!(config_file.is_file());
+
+        let config = read_config(config_root)
+            .context("Failed to read config")?
+            .expect("Config file does not exist in config root");
+        assert_eq!(config.network, KitsuneP2pConfig::mem());
+        assert!(matches!(
+            config.keystore,
+            KeystoreConfig::LairServerInProc { .. }
+        ));
```

Comment:
I suggest to assert the full config:
```suggestion
        let mut expected_root_path = root.clone().unwrap();
        expected_root_path.push(directory.unwrap());
        let mut expected_lair_root = expected_root_path.clone();
        expected_lair_root.push(KEYSTORE_DIRECTORY);
        let expected_admin_interface = AdminInterfaceConfig {
            driver: InterfaceDriver::Websocket {
                port: 0,
                allowed_origins: AllowedOrigins::Any,
            },
        };
        let expected_config = ConductorConfig {
            data_root_path: Some(expected_root_path.clone().into()),
            keystore: KeystoreConfig::LairServerInProc {
                lair_root: Some(expected_lair_root.into()),
            },
            admin_interfaces: Some(vec![expected_admin_interface]),
            ..Default::default()
        };
        assert_eq!(config, expected_config);
```